### PR TITLE
Float16 fixes

### PR DESCRIPTION
**Pull request by vcarpani**
This PR enables running with FP16.
Waiting on this PR on keras:
https://github.com/keras-team/keras/pull/11580

### CODE REVIEW

File: keras_retinanet/utils/image.py [hunk 1]
```diff
@@ -60,6 +63,20 @@ def preprocess_image(x, mode='caffe'):
     return x
 
 
+def cast_image_to_floatx(x):
```
Comment (hgaiser) [inline keras_retinanet/utils/image.py]: `keras.backend.cast_to_floatx` also works on np arrays

File: keras_retinanet/preprocessing/generator.py [hunk 2]
```diff
@@ -53,6 +54,7 @@ def __init__(
         compute_anchor_targets=anchor_targets_bbox,
         compute_shapes=guess_shapes,
         preprocess_image=preprocess_image,
+        cast_image_to_floatx=cast_image_to_floatx,
```
Comment (hgaiser) [inline keras_retinanet/preprocessing/generator.py]: I don't think we need to pass this to the generator as a functor.

**Comment by vcarpani** [comment]
@hgaiser the PR on keras is merged

File: keras_retinanet/utils/image.py (hunk 1 above)
Comment (hgaiser) [reply keras_retinanet/utils/image.py]: I think we can remove this wrapper entirely.

**Comment by hgaiser** [comment]
I have one comment, but after that it can be merged as far as I'm concerned.

**Pull request merged**